invenio/templates/configurations/logstash.yaml

```diff
@@ -52,67 +52,128 @@ data:
         }
-        ##############################################################
-        # USER AGENT
-        ##############################################################
-        # add ua/device only if set
-        if [@metadata][ua_device] {
-              mutate {
-                  add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
-              }
-          }
-        # add ua/family only if set
-        if [@metadata][ua_name] {
-            mutate {
-                add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
-            }
+        # Enrich with geolocation data.
+        geoip {
+            source => "http_x_forwarded_for" #field name that contains the ip address
+            fields => [ "city_name", "country_name", "country_code2", "continent_code"]
+            target => "data"
+            # Download static database and point to it
         }
-        # add ua/patch only if set
-        if [@metadata][ua_patch] {
-            mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
-            }
-        } else {
+
+      useragent {
+          source => "[data][user_agent]"
+          target => "[@metadata]"
+      }
+      ##############################################################
+      # USER AGENT
+      ##############################################################
+      # add ua/device only if set
+      if [@metadata][ua_device] {
             mutate {
-                add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+                add_field => { "[data][ua/device]" => "%{[@metadata][ua_device]}" }
             }
         }
-        # add ua/os only if set
-        if [@metadata][ua_os] {
-            mutate {
-                add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
-            }
+      # add ua/family only if set
+      if [@metadata][ua_name] {
+          mutate {
+              add_field => { "[data][ua/family]" => "%{[@metadata][ua_name]}" }
+          }
+      }
+      # add ua/patch only if set
+      if [@metadata][ua_patch] {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}.%{[@metadata][ua_patch]}" }
+          }
+      } else {
+          mutate {
+              add_field => { "[data][ua/full]" => "%{[@metadata][ua_name]}/%{[@metadata][ua_major]}.%{[@metadata][ua_minor]}" }
+          }
+      }
+      # add ua/os only if set
+      if [@metadata][ua_os] {
+          mutate {
+              add_field => { "[data][ua/os]" => "%{[@metadata][ua_os]}" }
+          }
+      }
+      # add a field with the category, based on the url
+      grok {
+         match => { "url" => ["^/(api/)?record/%{INT}/files/", "^/api/files/%{UUID}/", "^/api/deposit/depositions/%{INT}/files"]}
```

Comment:
Somme comments about this:

1. `record(s)`:
- old Zenodo URLs differ because of the `s` at the end:
  - `/record/<int>`: https://zenodo.org/record/8349181
  - `/api/records/<int>`: https://zenodo.org/api/records/8349181
- new URL is harmonized:
  - `/records/<string>`: https://inveniordm.web.cern.ch/records/cphz3-t9j70
  - `/api/records/<string>`: https://inveniordm.web.cern.ch/api/records/cphz3-t9j70

2. The recid of Zenodo is still an `int`, but any other RDM/new Invenio based repo use the new style `cool PIDs`, e.g. `/records/cphz3-t9j70`.
Can we relax this to string?

Reply:
Some patterns won't match in RDM either:

- `/api/deposit/depositions` is only valid for legacy Zenodo. There is no equivalent in RDM to fetch drafts, we only have `https://zenodo-rdm-qa.web.cern.ch/api/records`
- `/api/files/%{UUID}` won't work either. See https://zenodo-rdm-qa.web.cern.ch/api/records/10000019/files  

So I think that the only relevant pattern for `files` here would be:

`"^/(api/)?records/%{WORD}/files"` and maybe `^/api/records/%{WORD}/files-archive` (to download all the files of a record). I am not sure about needing the type `WORD` here

Reply:
The `%{WORD}` would not match nicola's example of `cphz3-t9j70`. Shall we put `%DATA` ? 
Let me know if this looks better

Reply:
@alejandromumo we need to keep both RDM and Zenodo legacy for the moment. Same for `/api/files...`

@psaiz yes, I fear that it is too greedy and the leading `/` will not work? to be tested.
https://github.com/elastic/elasticsearch/blob/main/libs/grok/src/main/resources/patterns/legacy/grok-patterns#L16

Reply:
Yeah either `DATA` or `GREEDYDATA` seem more adequate. `WORD` seems to match only text

Reply:
@ntarocco Zenodo legacy redirects files requests using flask redirect, therefore they will be a `HTTP 302 Found`. So I think we can drop legacy in this use case.

However, maybe we can add a new category `legacy` afterwards to understand whether these are being used or not. We know the routes that are redirected so we can add it later on. 

Is this somehow configurable?